libc/resolve_path.c

```diff
@@ -475,6 +475,48 @@ TEST(resolve_path, symlink_rename)
 }
 
 
+TEST(resolve_path, symlink_long_resolution)
+{
+	const char *dev = "/dev";
+	const char *pathSegment = "dev/../";
+	const char symNameShort[] = "symShort";
```

Comment:
Not doing a review, just curious - why is one a local array and two other pointers to strings?